projects/core/src/features-config/directives/feature-styles.service.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ComponentRef,
+  Injectable,
+  RendererFactory2,
+  inject,
+  isDevMode,
+} from '@angular/core';
+import { LoggerService } from '../../logger/logger.service';
+import { FeatureConfigService } from '../services/feature-config.service';
+
+@Injectable({ providedIn: 'root' })
+export class FeatureStylesService {
+  protected featureConfig = inject(FeatureConfigService);
+  private rendererFactory = inject(RendererFactory2); // private, because needed only to create a renderer
+  protected logger = inject(LoggerService);
```

Comment:
`nitpick`
```suggestion
  private rendererFactory = inject(RendererFactory2); // private, because needed only to create a renderer
  protected featureConfig = inject(FeatureConfigService);
  protected logger = inject(LoggerService);
```